src/sys/unix/waker.rs

```diff
@@ -148,26 +199,38 @@ mod pipe {
                 Err(ref err) if err.kind() == io::ErrorKind::WouldBlock => {
                     // The reading end is full so we'll empty the buffer and try
                     // again.
-                    self.empty();
+                    self.reset();
                     self.wake()
                 }
                 Err(ref err) if err.kind() == io::ErrorKind::Interrupted => self.wake(),
                 Err(err) => Err(err),
             }
         }
 
-        /// Empty the pipe's buffer, only need to call this if `wake` fails.
+        fn reset(&self) {
```

Comment:
Let's keep the original name, it's accurate, `reset`ting is not as the fd is not reset is any way.
```suggestion
        fn empty(&self) {
```